### ISSUE TITLE
Prefer_inline fix is overzealous

### ISSUE THREAD

**Issue by bjoern-jueliger-sap** [abaplint/abaplint#3033]
The prefer_inline check (justifiably) raises an issue for code like
```abap
data str type string.
str = '1 2 3'.
method( str ).
```
and the automated fix then turns this into 
```abap
data(str) = '1 2 3'.
method( str ).
```

The problem is that these two are not equivalent: In the inline-declared version, the inferred type of `str` is `c`, not `string`. This means that the second version has a syntax error when the input parameter of `method` has type `string`. 

The correct fully equivalent inline declaration would be either ``data(str) = `1 2 3`. `` or `data(str) = conv string( '1 2 3' ).`.

Additionally, the abaplint syntax check does not detect this kind of syntax error (but the rules for type inference for inline declarations are so arcane anyway that I didn't expect it to).

**Comment by larshp** [comment]
yea, the rule has some problems
